### PR TITLE
placement: add API support

### PR DESCRIPTION
**Pull request by disksing**
### What problem does this PR solve? 
continue merge placement rules.

### What is changed and how it works?
- add api support
- add raml document

### Check List 

Tests 
 - Manual test (add detailed scripts or steps below)

Code changes
 - Has HTTP API interfaces change

### CODE REVIEW

**Comment by disksing** [comment]
@lhy1024 I think we need the test. But I don't have enough time to add it now..

File: server/api/rule.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+}
+
+func (h *ruleHandler) GetAll(w http.ResponseWriter, r *http.Request) {
+	cluster := h.svr.GetRaftCluster()
+	if cluster == nil {
+		h.rd.JSON(w, http.StatusInternalServerError, server.ErrNotBootstrapped.Error())
+		return
+	}
+	if !cluster.IsPlacementRulesEnabled() {
+		h.rd.JSON(w, http.StatusPreconditionFailed, "placement rules feature is disabled")
```
Comment (nolouch) [inline server/api/rule.go]: How about defining an error?

Comment (disksing) [reply server/api/rule.go]: Good idea.

File: server/api/rule.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+func (h *ruleHandler) GetAllByGroup(w http.ResponseWriter, r *http.Request) {
+	cluster := h.svr.GetRaftCluster()
+	if cluster == nil {
+		h.rd.JSON(w, http.StatusInternalServerError, server.ErrNotBootstrapped.Error())
+		return
+	}
+	if !cluster.IsPlacementRulesEnabled() {
+		h.rd.JSON(w, http.StatusPreconditionFailed, placementDisabledErr.Error())
+		return
+	}
```
Comment (shafreeck) [inline server/api/rule.go]: It is really verbose of these lines of code. Maybe you want to try the middleware https://github.com/gorilla/mux#middleware

File: server/api/rule.go [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+	keyHex := mux.Vars(r)["key"]
+	key, err := hex.DecodeString(keyHex)
+	if err != nil {
+		h.rd.JSON(w, http.StatusBadRequest, "key should be hex format")
```
Comment (shafreeck) [inline server/api/rule.go]:
```suggestion
		h.rd.JSON(w, http.StatusBadRequest, "key should be in hex format")
```

File: server/api/rule.go [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+		h.rd.JSON(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+	h.rd.JSON(w, http.StatusOK, nil)
+}
+
+func (h *ruleHandler) checkRule(r *placement.Rule) error {
+	start, err := hex.DecodeString(r.StartKeyHex)
+	if err != nil {
+		return errors.Wrap(err, "start key is not hex format")
```
Comment (shafreeck) [inline server/api/rule.go]:
```suggestion
		return errors.Wrap(err, "start key is not in hex format")
```

File: server/api/rule.go (hunk 2 above)
Comment (disksing) [reply server/api/rule.go]: I'll see what I can do for now.

**Pull request merged**